runtime/util/vmargs.c

```diff
@@ -1496,11 +1499,18 @@ addLauncherArgs(J9PortLibrary * portLib, JavaVMInitArgs *launcherArgs, UDATA lau
 		}
 
 		if (TRUE == convertEncoding) {
-			optArg = newJavaVMArgInfo(vmArgumentsList, transcodedString, ARG_MEMORY_ALLOCATION|consumableFlag);
+			if ((strncmp(optString, "-Djava.class.path=", strlen("-Djava.class.path=")) != 0) && (strncmp(optString, "-Dsun.java.command=", strlen("-Dsun.java.command=")) != 0) && (strncmp(optString, "-Dsun.java.launcher=", strlen("-Dsun.java.launcher=")) != 0) && (strncmp(optString, "-Dsun.java.launcher.pid=", strlen("-Dsun.java.launcher.pid=")) != 0)) {
```

Comment:
Not too sure on how I can exclude the default options in this if statement. Does anyone have any ideas?

Reply:
I don't think this will catch all the system arguments based on the output from #2974 such as `-Xjcl:jclse11_29`. what was your command line output?

Reply:
do you have any ideas for doing this other than hard coding @pdbain-ibm ?

Reply:
Without the changes, the output for me was:

-Xoptionsfile=/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/lib/options.default, -Xlockword:mode=default,noLockword=java/lang/String,noLockword=java/util/MapEntry,noLockword=java/util/HashMap$Entry,noLockword=org/apache/harmony/luni/util/ModifiedMap$Entry,noLockword=java/util/Hashtable$Entry,noLockword=java/lang/invoke/MethodType,noLockword=java/lang/invoke/MethodHandle,noLockword=java/lang/invoke/CollectHandle,noLockword=java/lang/invoke/ConstructorHandle,noLockword=java/lang/invoke/ConvertHandle,noLockword=java/lang/invoke/ArgumentConversionHandle,noLockword=java/lang/invoke/AsTypeHandle,noLockword=java/lang/invoke/ExplicitCastHandle,noLockword=java/lang/invoke/FilterReturnHandle,noLockword=java/lang/invoke/DirectHandle,noLockword=java/lang/invoke/ReceiverBoundHandle,noLockword=java/lang/invoke/DynamicInvokerHandle,noLockword=java/lang/invoke/FieldHandle,noLockword=java/lang/invoke/FieldGetterHandle,noLockword=java/lang/invoke/FieldSetterHandle,noLockword=java/lang/invoke/StaticFieldGetterHandle,noLockword=java/lang/invoke/StaticFieldSetterHandle,noLockword=java/lang/invoke/IndirectHandle,noLockword=java/lang/invoke/InterfaceHandle,noLockword=java/lang/invoke/VirtualHandle,noLockword=java/lang/invoke/PrimitiveHandle,noLockword=java/lang/invoke/InvokeExactHandle,noLockword=java/lang/invoke/InvokeGenericHandle,noLockword=java/lang/invoke/VarargsCollectorHandle,noLockword=java/lang/invoke/ThunkTuple, -Xjcl:jclse29, 
-Dcom.ibm.oti.vm.bootstrap.library.path=/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/lib/compressedrefs:/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/lib, -Dsun.boot.library.path=/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/lib/compressedrefs:/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/lib, -Djava.library.path=/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/lib/compressedrefs:/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/lib:/usr/lib64:/usr/lib, -Djava.home=/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk, 
-Duser.dir=/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/bin, 
--add-exports=java.base/com.ibm.oti.vm=ALL-UNNAMED, 
-Djava.class.path=., 
-Dsun.java.command=Test, 
-Dsun.java.launcher=SUN_STANDARD, 
-Dsun.java.launcher.pid=49456]

Reply:
right so those extra few 
`-Xoptionsfile=/root/openj9-openjdk-jdk11/build/linux-x86_64-normal-server-release/images/jdk/lib/options.default,
-Xlockword:mode=default,noLockword=java/lang/String,noLockword=java/util/MapEntry,noLockword=java/util/HashMap$Entry,noLockword=org/apache/harmony/luni/util/ModifiedMap$Entry,noLockword=java/util/Hashtable$Entry,noLockword=java/lang/invoke/MethodType,noLockword=java/lang/invoke/MethodHandle,noLockword=java/lang/invoke/CollectHandle,noLockword=java/lang/invoke/ConstructorHandle,noLockword=java/lang/invoke/ConvertHandle,noLockword=java/lang/invoke/ArgumentConversionHandle,noLockword=java/lang/invoke/AsTypeHandle,noLockword=java/lang/invoke/ExplicitCastHandle,noLockword=java/lang/invoke/FilterReturnHandle,noLockword=java/lang/invoke/DirectHandle,noLockword=java/lang/invoke/ReceiverBoundHandle,noLockword=java/lang/invoke/DynamicInvokerHandle,noLockword=java/lang/invoke/FieldHandle,noLockword=java/lang/invoke/FieldGetterHandle,noLockword=java/lang/invoke/FieldSetterHandle,noLockword=java/lang/invoke/StaticFieldGetterHandle,noLockword=java/lang/invoke/StaticFieldSetterHandle,noLockword=java/lang/invoke/IndirectHandle,noLockword=java/lang/invoke/InterfaceHandle,noLockword=java/lang/invoke/VirtualHandle,noLockword=java/lang/invoke/PrimitiveHandle,noLockword=java/lang/invoke/InvokeExactHandle,noLockword=java/lang/invoke/InvokeGenericHandle,noLockword=java/lang/invoke/VarargsCollectorHandle,noLockword=java/lang/invoke/ThunkTuple, -Xjcl:jclse29,
`
will need excluding as well

Reply:
Right, with my changes the extra few options are already excluded (since those arguments are not added to the the argument array (`vm_args`) when `addLauncherArgs()` is called. It's just that unfortunately, the arguments that are added to `vm_args` before `addLauncherArgs()` is called, are all reported with my current implementation.

Reply:
Perhaps if anyone could help me identify where the arguments (`-Djava.class`...etc.) are added, I might be able to find a better solution. But ATM, the only solution I have is hard coding it.

Reply:
> do you have any ideas for doing this other than hard coding @pdbain-ibm ?

Not that I can think of just now.  These come from the Java launcher, which comes from OpenJDK.

You should also include arguments from `IBM_JAVA_OPTIONS` and `JAVA_TOOL_OPTIONS`

Reply:
I was thinking of creating an "Ignored  user options table" (similiar to `J9VMIgnoredOption ignoredOptionTable[]` in jvminit.c) where arguments in the table are parsed beforehand so that they are not returned. While is is still "hard coded", it's a lot more elegant than our current implementation...

Reply:
@pdbain I've pushed changes to include arguments from `IBM_JAVA_OPTIONS` and `JAVA_TOOL_OPTIONS` :)

Reply:
Sounds like a good plan.  Keep it in sorted order so you can search in O(lg N).